Sources/HiPMobile/BusinessLayer/Managers/DbManager.cs

```diff
@@ -12,85 +12,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-using JetBrains.Annotations;
-using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Models;
+using PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.DtoToModelConverters;
 using PaderbornUniversity.SILab.Hip.Mobile.Shared.Common;
 using PaderbornUniversity.SILab.Hip.Mobile.Shared.DataAccessLayer;
-using Realms;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace PaderbornUniversity.SILab.Hip.Mobile.Shared.BusinessLayer.Managers
 {
     /// <summary>
-    /// Access to database methods.
+    /// Provides access to database methods.
     /// </summary>
     public static class DbManager
     {
-        private static readonly IDataAccess DataAccess = IoCManager.Resolve<IDataAccess>();
-
-        /// <summary>
-        /// Creates an object of type T that is synced to the database.
-        /// </summary>
-        /// <typeparam name="T">The type of the object being created. T needs to be subtype of RealmObject and implement the IIdentifiable interface.</typeparam>
-        /// <returns>The instance.</returns>
-        public static T CreateBusinessObject<T>() where T : RealmObject, IIdentifiable, new()
-        {
-            return DataAccess.CreateObject<T>();
-        }
-        
         /// <summary>
-        /// Creates an object of type T that is synced to the database.
+        /// Provides read-only database access. To modify the database in any way,
+        /// use <see cref="StartTransaction(IEnumerable{object})"/> instead.
         /// </summary>
-        /// <param name="id">The ID to assign to the object.</param>
-        /// <param name="updateCurrent">If true, first removes any object of the same type with the id.</param>
-        /// <typeparam name="T">The type of the object being created. T needs to be subtype of RealmObject and implement the IIdentifiable interface.</typeparam>
-        /// <returns>The instance.</returns>
-        public static T CreateBusinessObject<T>([NotNull] string id, bool updateCurrent = false) where T : RealmObject, IIdentifiable, new()
-        {
-            return DataAccess.CreateObject<T>(id, updateCurrent);
-        }
+        public static IReadOnlyDataAccess DataAccess { get; } = IoCManager.Resolve<IDataAccess>();
 
         /// <summary>
-        /// Delete an object of type T from the database.
+        /// Starts a new write transaction. Make sure to close the transaction by either committing it or rolling back.
```

Comment:
`using` calls `Dispose()` automatically which, in our transaction implementation, commits the transaction. So to be precise, only `Rollback()` needs to be called explicitly, if needed. Although I strongly disagree with this, I didn't want to invert our transaction semantics and confuse everyone, so I left everything as is, including this comment.

The problem with `Dispose()` automatically committing the transaction is that if an exception occurs within the transaction scope, one would expect that the transaction is rolled back (instead of committing the probably inconsistent changes)

Reply:
So my suggestion would be to make `Dispose()` call `Rollback()` (if the transaction hasn't yet been committed or rolled back) and force devs to call `Commit()` explicitly. But changing the semantics is probably enough work for a separate task.

Reply:
I see. An even better approach would probably be to pass a lambda to a function that executes the lambda in a transaction and commits or rolls back when an exception occurs. If you agree, I'll create a separate issue for this on JIRA.

Reply:
I think I'd prefer to keep the `using`-approach because lambdas have a few limitations. For example, you can use `return` in a `using`-block to immediately exit the whole method. You can't do this with the lambda approach. But I'd be fine with both approaches and it's definitely worth a JIRA issue.